packages/openspp/src/Adaptor.js

```diff
@@ -660,6 +658,311 @@ export function unenroll(registrant_id, program_id) {
   };
 }
 
+/**
+ * create new individual for OpenSPP
+ * @public
+ * @example
+ * createIndividual({ name: "Individual 1" })
+ * @function
+ * @param {object} data - registrant create data
+ * @param {function} callback - An optional callback function
+ * @returns {Operation}
+ */
+export function createIndividual(data, callback=false) {
+  return async state => {
+    try {
+      if (typeof data !== 'object' || Array.isArray(data) || data === null) {
```

Comment:
Do you really need or want to do this kind of validation in the adaptor?

If someone passes invalid data, won't the OpenSPP backend throw a userful error?

You can leave this stuff if you want, but my advice is to leave validation to the backend as much as possible. Adaptor-side validation should probably only focus on very common use-cases.

Otherwise you can end up writing validation code forever and it's all sort of redundant and might not even be useful.

Reply:
done removing :D